src/ast/element-descriptor.ts

```diff
@@ -38,24 +38,25 @@ export class ScannedElement implements ScannedFeature, Resolvable {
   description = '';
   demos: {desc?: string; path: string}[] = [];
   events: ScannedEvent[] = [];
-  sourceLocation: SourceLocation;
   node: estree.Node;
+  sourceRange: SourceRange;
 
   jsdoc?: jsdoc.Annotation;
 
   applyLocationOffset(locationOffset?: LocationOffset) {
     if (!locationOffset) {
       return;
     }
-    this.sourceLocation =
-        correctSourceLocation(this.sourceLocation, locationOffset);
+    this.sourceRange = correctSourceRange(this.sourceRange, locationOffset);
```

Comment:
what would it take to push this upstream to be correct when the object is created? The parser needs to know the loationOffset probably...